### PR TITLE
[PyTorch] Add support for aten.neg

### PR DESCRIPTION
**Pull request by Atrisan**
## Description
Add neg libnode for integer negation which is not supported as a native tasklet.
Add support for aten.neg

## Related Issues

- Closes #

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor / internal change (no functional change)
- [ ] Documentation update

## Quality Checklist
*Please ensure the following are completed before requesting review:*

* [x] Runtime-compatible (no externally visible existing method has its parameters changed, was deleted or has its exception behavior changed)
* [x] Compile-time compatible (new arguments & properties added have default values, but users need to recompile against the new version to work)

IR:
* [ ] Introduces new types into the SDFG IR
* [x]  Changed / added properties on SDFG IR Elements (LibraryNodes, ControlFlowNodes DataFlowNodes etc.)
  *  [x] New / modified types have serialization / deserialization support
  *  [x] Deserialization is backward compatible (new serialization can deserialize old format and represent it in the new format if applicable)
  *  [x] Serialized format is backward compatible (previous deserializer can read new serialized format without incorrectness (just lacking additional guarantees that are optional or did not exist before)
  *  [x] Serializer has an option to emit the previous format

### Unit Tests
- [x] New unit tests have been added for the changes.
- [x] Existing unit tests pass locally.
- [x] Edge cases and error paths are covered.

### Integration Tests
- [x] New integration tests have been added (or existing ones updated).
- [x] Integration tests pass locally.
- [x] Interactions with other components/services are verified.

### Documentation
- [ ] Public APIs, options, and behavior changes are documented.
- [x] README / docs site / inline comments updated where applicable.
- [ ] Changelog / release notes updated (if applicable).

### Test Coverage
- [x] Test coverage has not decreased as a result of this change.
- [x] Coverage report reviewed and acceptable.
- [x] Critical / new code paths are covered.

## How Has This Been Tested?
Describe the tests you ran to verify your changes and provide instructions to reproduce.
Include relevant details for your test configuration (OS, Python/LLVM/Docc version, hardware/target).

### CODE REVIEW

**Comment by Moehre2** [comment]
Im not sure if we need a tensor node for negation... This can be done with tasklet tensor node and fp_neg or int_sub (and 0)

**Comment by Atrisan** [comment]
For fp yes, and thats what I do.
For int its undefined. So int_sub or int_mul still require an additional constant input. A Tasklet parser cannot do this.

File: pytorch/docc/pytorch/graph_parser/elementwise.py [hunk 1]
```diff
@@ -72,6 +72,158 @@ def parse(
...
+    ) -> None:
+        if len(node.args) != 1:
+            raise GraphParserError(
+                self,
+                node,
+                "Expected exactly one argument but got " + str(len(node.args)),
+            )
+        self_container: str = self.get_arg_container(node, container_info, 0)
+        self_tensor: Tensor = self.get_tensor_type(node, container_info, self_container)
+        primitive_type = self_tensor.element_type.primitive_type
```
Comment (WashimNeupane) [inline pytorch/docc/pytorch/graph_parser/elementwise.py:211]: [get_arg_container] and [get_tensor_type] are called here for dispatch, then called again inside the sub-parser. An alternative is to dispatch directly from torch fx metadata
`dtype = node.args[0].meta["val"].dtype`
`if dtype.is_floating_point:`

File: pytorch/docc/pytorch/graph_parser/elementwise.py [hunk 2]
```diff
@@ -72,6 +72,158 @@ def parse(
...
+        self.fp_parser: GraphParserModule = fp_parser
+        self.int_parser: GraphParserModule = int_parser
+
+    def parse(
+        self,
+        node: torch.fx.Node,
+        builder: StructuredSDFGBuilder,
+        container_info: ContainerInfos,
+    ) -> None:
+        if len(node.args) != 1:
```
Comment (WashimNeupane) [inline pytorch/docc/pytorch/graph_parser/elementwise.py:199]: redundant check.

**Review by WashimNeupane — CHANGES_REQUESTED** [review]

File: pytorch/docc/pytorch/graph_parser/elementwise.py (hunk 1 above)
Comment (Atrisan) [reply pytorch/docc/pytorch/graph_parser/elementwise.py:211]: I'd rather stay in the parser infrastructure that is available, than use the torch API in more places than necessary to avoid larger rewrites on API changes.

File: pytorch/docc/pytorch/graph_parser/elementwise.py (hunk 2 above)
Comment (Atrisan) [reply pytorch/docc/pytorch/graph_parser/elementwise.py:199]: technically yes, but it helps with debugging and ensures we actually have an argument for the type checking.
I added the kwargs check as well
